src/common/variables.c

```diff
@@ -555,15 +555,46 @@ static char *_get_base_value(dt_variables_params_t *params, char **variable)
   else if(_has_prefix(variable, "SEQUENCE"))
   {
     uint8_t nb_digit = 4;
+    guint shift = 1;
+    // old $(SEQUENCEn) syntax for backwards compatibility
     if(g_ascii_isdigit(*variable[0]))
     {
       nb_digit = (uint8_t)*variable[0] & 0b1111;
       (*variable) ++;
     }
+    // new $(SEQUENCE[n,m]) syntax
```

Comment:
I think we should have a generic routine to parse the list of parameters. No need to do something fancy, speed is not an issue here anyway. So what about a routine returning a list of string one for each parameter. Then in the `SEQUENCE` and `CATEGORY` section one just need to iterate over this list and do whatever is needed (possibly converting to an int for example).

So I would check for `[` and if found use the new support otherwise keep the current parsing for `SEQUENCE` and `CATEGORY`.

With this routine it will be easier to add support for parameters for more variables if needed later.

Reply:
done